src/proc_fuse.c

```diff
@@ -213,6 +213,24 @@ static uint64_t get_memlimit(const char *cgroup, bool swap)
 	return memlimit;
 }
 
+/*
+ * dirname() behaves differently on musl than glibc, so implement the bits we
+ * need (the path in this case is always absolute and never ends in a `/`)
+ */
+static char *cgroup_parent(char *cgroup)
+{
+	char *last_slash = NULL;
+	if (strcmp(cgroup, "/") == 0)
+		return cgroup;
+
+	last_slash = strrchr(cgroup, '/');
+	if (last_slash == cgroup)
+		last_slash++;
+	*last_slash = '\0';
+
+	return cgroup;
+}
```

Comment:
Hm, I think this doesn't really work. What do you expect the output of:
"foo/bar" to be for `cgroup_parent()` vs `dirname()`?
`dirname(foo/bar)` gives you "foo" but `cgroup_parent()` gives you "foo/bar".

Reply:
Actually I think `cgroup_parent()` does give you `"foo"` in that case? Where `cgroup_parent()` breaks down (most critically, there are definitely more edge cases that GNU `dirname()` covers) is with `cgroup_parent("foo")`, which segfaults. I could add a check for this but in the use case of `get_min_memlimit()` (as far as I can tell) the path will always start with a `/`. The `strcmp(copy, "/") == 0` in `gen_min_memlimit()` would break in the case of a path like `"foo"` anyway?

Am I right in thinking the value is only ever the third item in a `/proc/*/cgroup` tuple (e.g. `/user.slice/user-1000.slice/session-1.scope`?)